packages/temporal/workflows/mtv.go

```diff
@@ -148,9 +148,66 @@ func MtvRoomWorkflow(ctx workflow.Context, params shared.MtvRoomParameters) erro
 			},
 		},
 
-		Initial: MtvRoomPausedState,
+		Initial: MtvRoomFetchInitialTracks,
 
 		States: brainy.StateNodes{
+
+			MtvRoomFetchInitialTracks: &brainy.StateNode{
+				OnEntry: brainy.Actions{
+					brainy.ActionFn(
+						func(c brainy.Context, e brainy.Event) error {
+
+							ao := workflow.ActivityOptions{
+								ScheduleToStartTimeout: time.Minute,
+								StartToCloseTimeout:    time.Minute,
+							}
+							ctx = workflow.WithActivityOptions(ctx, ao)
+
+							fetchedInitialTracksFuture = workflow.ExecuteActivity(
+								ctx,
+								activities.FetchTracksInformationActivity,
+								internalState.TracksIDsList,
+							)
+
+							return nil
+						},
+					),
+				},
+
+				On: brainy.Events{
+					MtvRoomInitialTracksFetched: brainy.Transitions{
+						{
+							Target: MtvRoomPausedState,
+
+							Actions: brainy.Actions{
+								brainy.ActionFn(
+									func(c brainy.Context, e brainy.Event) error {
+
+										event := e.(MtvRoomInitialTracksFetchedEvent)
+										internalState.Tracks = event.Tracks
+
+										if tracksCount := len(event.Tracks); tracksCount > 0 {
+											currentTrack := internalState.Tracks[0]
+											internalState.CurrentTrack = currentTrack
+
+											if tracksCount == 1 {
+												internalState.Tracks = []shared.TrackMetadata{}
+												internalState.TracksIDsList = []string{}
+											} else {
+												internalState.Tracks = internalState.Tracks[1:]
+												internalState.TracksIDsList = internalState.TracksIDsList[1:]
+											}
+										}
+
+										return nil
+									},
+								),
+							},
+						},
+					},
```

Comment:
As there is only one possible transition for this event, we can simplify the code. It absolutely does not change the behaviour:

```suggestion
					MtvRoomInitialTracksFetched: brainy.Transition{
						Target: MtvRoomPausedState,

						Actions: brainy.Actions{
							brainy.ActionFn(
								func(c brainy.Context, e brainy.Event) error {

								event := e.(MtvRoomInitialTracksFetchedEvent)
								internalState.Tracks = event.Tracks

									if tracksCount := len(event.Tracks); tracksCount > 0 {
										currentTrack := internalState.Tracks[0]
										internalState.CurrentTrack = currentTrack

										if tracksCount == 1 {
											internalState.Tracks = []shared.TrackMetadata{}
											internalState.TracksIDsList = []string{}
										} else {
											internalState.Tracks = internalState.Tracks[1:]
											internalState.TracksIDsList = internalState.TracksIDsList[1:]
										}
									}

									return nil
								},
							),
						},
					},
```

Reply:
Fixed !